superset/assets/src/explore/components/ControlPanelsContainer.jsx

```diff
@@ -89,17 +89,23 @@ class ControlPanelsContainer extends React.Component {
   }
   render() {
     const allSectionsToRender = this.sectionsToRender();
+    const controlOverrides = visTypes[this.props.controls.viz_type.value].controlOverrides || {};
     const querySectionsToRender = [];
     const displaySectionsToRender = [];
     allSectionsToRender.forEach((section) => {
       if (section.controlSetRows.some(rows => rows.some(
-        control => (
-          controls[control] &&
-          (
-            !controls[control].renderTrigger ||
-            controls[control].tabOverride === 'data'
-          )
-        )))) {
+        (control) => {
```

Comment:
This code section is becoming even more opaque than it was before, and it was already bad before... For instance I'm confused between what's a control name and a control object, or object of objects, ... 

Also confused why we need to apply the overrides here where it should all be done in one place.

Reply:
I just checked and it appears the control overrides happen early on in `getControlsState`, why do we need it here too?

Reply:
hi, @mistercrunch, querySectionsToRender and displaySectionsToRender is dependent on renderTrigger/tabOverride properties in every control object,
before implement control override in VizType can not do overide here.
you can test this in Histogram diagram